test/visual/mpl/circuit/test_circuit_matplotlib_drawer.py

```diff
@@ -49,8 +50,11 @@
 else:
     raise ImportError('Must have Matplotlib installed. To install, run "pip install matplotlib".')
 
-
-RESULTDIR = os.path.dirname(os.path.abspath(__file__))
+BASEDIR = os.path.dirname(os.path.abspath(__file__))
+RESULTDIR = os.path.join(BASEDIR, "circuit_results", "")
+TEST_REFERENCE_PATH = os.path.join(BASEDIR, "references")
+FAILURE_DIFF_DIR = os.path.join(os.path.dirname(BASEDIR), "visual_test_failures", "")
+FAILURE_PREFIX = "circuit_failure_"
```

Comment:
We seem to be inconsistent on whether directories are referred to as "DIR" or "PATH", and whether we use shouting snake case, or all-caps one-word.  Could we choose one for each, please?

Minor style choice, but these might be better off as `pathlib.Path` instances, then joining paths is much easier - `DIR / "my_file.png"` - to the degree that I think the `_image_path` etc methods could be skipped.  That would make the test logic a bit easier to read for me - there's lots of indirection at the moment that I have to keep in my head.

Reply:
Changed consistency to shouting snake case `DIR`.

Changed to path lib.